docs/layout-of-source-files.rst

```diff
@@ -216,101 +216,562 @@ the code below creates new global symbols ``alias`` and ``symbol2`` which refere
+   are not affected.
+
+    .. code-block:: bash
+
+        solc /project=/contracts /project/contract.sol # Import key: /project/contract.sol
+
+#. **Context and prefix must match import keys, not import paths.**
+
+   - This means that you cannot remap ``./`` or ``../`` directly since they are replaced during
+     translation to import keys but you can remap the source locations they resolve into:
+
+     .. code-block:: bash
+
+         solc ./=a /project=b /project/contract.sol
+
+     .. code-block:: solidity
+         :caption: /project/contract.sol
+
+         import "./util.sol" as util; // Import key: b/util.sol
+
+   - You cannot remap base path or any other part of the path that is only added when the file is
+     looked up on the underlying filesystem.
+
+     .. code-block:: bash
+
+         solc /project=/contracts /project/contract.sol --base-path /project
+
+     .. code-block:: solidity
+         :caption: /project/contract.sol
+
+         import "util.sol" as util; // Import key: util.sol
 
-As a more complex example, suppose you rely on a module that uses an old
-version of dapp-bin that you checked out to ``/usr/local/dapp-bin_old``, then you can run:
+#. **Target is inserted directly into the import key and does not necessarily have to be a valid path.**
+
+   - It can be anything as long as the file loader can handle it.
+     In case of the command-line interface this includes also relative paths.
+     When using the JavaScript interface you can just as well use URLs and abstract identifiers if
+     your callback can handle them.
+
+   - Remapping happens after paths relative to the source directory have already been resolved.
+     This means that targets starting with ``./`` and ``../`` have no special meaning and are
+     relative to the base directory rather than to the source location.
+
+   - Remapping targets are not normalized so ``@root=./a/b//`` will remap ``@root/contract.sol``
+     to ``./a/b//contract.sol`` and not ``a/b/contract.sol``.
+
+   - If the target does not end with a slash, the compiler will not add one automatically:
+
+     .. code-block:: bash
+
+         solc /project/=/contracts /project/contract.sol
+
+     .. code-block:: solidity
+         :caption: /project/contract.sol
+
+         import "/project/util.sol" as util; // Import key: /contractsutil.sol
+
+#. **Context and prefix are patterns and matches must be exact.**
+
+   - ``a//b=c`` will not match ``a/b``.
+
+   - Import keys are not normalized so ``a/b=c`` will not match ``a//b`` either.
+
+   - Parts of file and directory names can match as well.
+     ``/newProject/con:/new=old`` will match ``/newProject/contract.sol`` and remap it to
+     ``oldProject/contract.sol``.
+
+#. **At most one remapping can be applied to a single import.**
+
+    - If multiple remappings match the same key, the one with the longest matching prefix is chosen.
+    - If prefixes are identical, the one specified last wins.
```

Comment:
I think that using duplicate prefixes with the same context should be an error. It should still be possible to have identical prefixes in different contexts though.

I haven't checked what happens in Standard JSON when there are multiple identical prefixes but I'm assuming it's the same as on the command line (I will need to include that case in tests).